syntax = "proto3";

package chat;


service Chat {
	rpc ChatStream(stream StreamRequest) returns  (stream StreamResponse) {}
}


message StreamRequest {
    oneof RequestEvent {
        StartSession startSession_request = 1;
        JoinSession joinSession_request = 2;
		Message  message_request  = 3;
        QuitSession quitSession_request = 4;
    }
}


message StreamResponse {
     oneof ResponseEvent {
        Login    login_response    = 1;
        Logout   logout_response   = 2;
        Message  message_response  = 3;
		StartSessionResponse startsession_response = 4;
		JoinSessionResponse joinsession_response = 5;
    }
}


message StartSession {
    string userName = 1;
}

message JoinSession {
    string userName = 1;
    string session_id = 2;
}
	
message Message {
    string userName = 1;
	string session_id = 2;
    string message = 3;
}
	
message QuitSession {
    string userName = 1;
	string session_id = 2;
}
	
message Login {
    string userName = 1;
}

message Logout {
    string userName = 1;
}

message StartSessionResponse {
    string session_id = 1;
}

message JoinSessionResponse {
    string message = 1;
}

